HW_6/task_2.js

```diff
@@ -0,0 +1,271 @@
+    const maxDeptId = sumIds();           
+    let newDep = { name: nameDepartment, id: maxDeptId + 1, employees_count: 0 };
+    departments.push(newDep);
+    return newDep;
+};
+
+console.log(addDepartmentToEnterprise(11, 'Автотестеры'));
+
+// 5. Написать функцию для редактирования названия предприятия. Принимает в качестве аргумента id предприятия и новое имя предприятия.
+
+// Пример:
+// editEnterprise(1, "Новое название предприятия")
+
+function editEnterprise(id, newFirmName){
+    getEnterpriseByIdOrName(id).name = newFirmName;    
+    return getEnterpriseByIdOrName(id).name;
+};
+
+console.log(editEnterprise(11, "Предприятие 4"));
+
+
+// 6. Написать функцию для редактирования названия отдела. Принимает в качестве аргумента id отдела и новое имя отдела.
+
+// Пример:
+// editDepartment(7, "Новое название отдела")
+
+function editDepartment(departmentId, newDepartmentName) {
+    let departments = getEnterpriseByDepartment(departmentId).departments;
+    let foundDeparment = departments.find((department) => department.id === departmentId);
+    foundDeparment.name = newDepartmentName;
+    return foundDeparment    
+}
+
+console.log(editDepartment(12, "AQA"))
+
+// 7. Написать функцию для удаления предприятия. В качестве аргумента принимает id предприятия.
+
+// Пример:
+// deleteEnterprise(1)
+
+function deleteEnterprise(enterpriseID) {    
+    const index = enterprises.findIndex(enterprise => enterprise.id === enterpriseID);
+    if (index !== -1) {
+        enterprises.splice(index, 1);
+    }
+    return enterprises;
+}
+
+console.log(deleteEnterprise(11))
+
+
+// 8. Написать функцию для удаления отдела. В качестве аргумента принимает id отдела. Удалить отдел можно только, если в нем нет сотрудников.
+
+// Пример:
+// deleteDepartment(3)
+
+function deleteDepartment(departmentId) {
+    let departments = getEnterpriseByDepartment(departmentId).departments;    
+        const departmentIndex = departments.findIndex(department => department.id === departmentId && department.employees_count === 0);
+        if (departmentIndex !== -1) {
+            departments.splice(departmentIndex, 1);
+        };
+
+    return enterprises;
+    };
+    
+
+console.log(deleteDepartment(10));
+
+// 9. Написать функцию для переноса сотрудников между отделами одного предприятия. 
+//В качестве аргумента принимает два значения: id отдела, из которого будут переноситься сотрудники и id отдела, в который будут переноситься сотрудники).
+
+// Пример:
+// moveEmployees(2, 3)
+
+function moveEmployees(idMoveFrom, idMoveTo) {
+    let departmentFrom = null;
+    let departmentTo = null;
+    
+    enterprises.forEach(enterprise => {
+        if (!departmentFrom) {
+            departmentFrom = enterprise.departments.find(department => department.id === idMoveFrom);
+        }
+        if (!departmentTo) {
+            departmentTo = enterprise.departments.find(department => department.id === idMoveTo);
+        }
+    });

```

Comment:
You never get `true` for `departmentFrom or departmentTo`
```suggestion
    enterprises.forEach(enterprise => {
            departmentFrom = enterprise.departments.find(department => department.id === idMoveFrom);
            departmentTo = enterprise.departments.find(department => department.id === idMoveTo);

    });

```